next/broker/celery_app/tasks.py

```diff
@@ -7,20 +7,58 @@
 import json
 import traceback
 import numpy
-from next.constants import DEBUG_ON
 import hashlib
 
-# import next.logging_client.LoggerHTTP as ell
-from next.database_client.DatabaseAPI import DatabaseAPI
-db = DatabaseAPI()
-from next.logging_client.LoggerAPI import LoggerAPI
-ell = LoggerAPI()
 import next.utils
 import next.constants
-import next.apps.Butler as Butler
+from next.constants import DEBUG_ON
+
+from next.database_client.DatabaseAPI import DatabaseAPI, DatabaseException
+from next.logging_client.LoggerAPI import LoggerAPI
+
+from next.apps.Butler import Butler
 import next.lib.pijemont.verifier as verifier
 
-Butler = Butler.Butler
+db, ell = None, None
+
+def worker_connect_db():
+    global db, ell
```

Comment:
Ah, I see what you mean. Move the `global` statements to the Celery hooks and have `worker_connect_db()` return `db, ell`. I'll do that; it's far more elegant.

As far as connection failures, the [code will attempt to reconnect](https://github.com/nextml/NEXT/blob/21c6ce1f62bb65ef0bb8ad4c9e6118efad847e08/next/broker/celery_app/tasks.py#L28-L35), doubling the backoff on each failure. If a Celery worker runs initialization for too long, Celery will entirely drop the worker and retry.
Returning `None` from the `DatabaseAPI` constructor would cause piles of upstream error `NoneType has no attribute $blah` messages, which is why I decided to bubble up connection errors as a `DatabaseError` as suggested by, iirc, Devin.